ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/CategoryController.java

```diff
@@ -0,0 +1,138 @@
+package com.metaphorce.exam.ecommerce.product.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.metaphorce.exam.ecommerce.product.dto.CategoryDTO;
+import com.metaphorce.exam.ecommerce.product.mapper.CategoryMapper;
+import com.metaphorce.exam.ecommerce.product.model.impl.Category;
+import com.metaphorce.exam.ecommerce.product.service.impl.CategoryService;
+
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/category")
+@RestController
+@Slf4j
+public class CategoryController {
+	@Autowired
+	private CategoryService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Category
+	 * @param Category Data to insert
+	 * @return New Category create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<CategoryDTO> createCategory(@Valid @RequestBody CategoryDTO CategoryRequest) {
+		log.info("Category to create: " + CategoryRequest);
+		ResponseEntity<CategoryDTO> response = null;
+		Category newCategory = this.service.create(CategoryMapper.mapper.toEntity(CategoryRequest));
+
+		if(newCategory != null) {
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.